crates/cxx-qt-gen/src/naming/type_names.rs

```diff
@@ -168,18 +159,24 @@ impl TypeNames {
         }
 
         for extern_cxxqt in &cxx_qt_data.extern_cxxqt_blocks {
-            // TODO: Refactor, this is a hack to reconstruct the original ItemForeignMod
-            let foreign_mod = ItemForeignMod {
-                attrs: extern_cxxqt.attrs.clone(),
-                unsafety: None,
-                brace_token: Brace::default(),
-                items: extern_cxxqt.passthrough_items.clone(),
-                abi: syn::Abi {
-                    extern_token: syn::parse_quote!(extern),
-                    name: None,
-                },
+            let namespace = if let Some(namespace) = &extern_cxxqt.namespace {
+                quote! { #[namespace = #namespace ] }
+            } else {
+                quote! {}
+            };
+
+            let items = extern_cxxqt.passthrough_items.clone();
+            let foreign_mod: ItemForeignMod = parse_quote! {
+                #namespace
+                extern "C++" {
+                    #(#items)*
+                }
             };
             self.populate_from_foreign_mod_item(&foreign_mod, bridge_namespace, module_ident)?;
+
+            for qobject in extern_cxxqt.qobjects.iter() {
+                self.insert(qobject.name.clone())?;
+            }
```

Comment:
I moved qobjects out of the passthrough_items vector since it impacted how I could rebuild them, so they then have to be added to type_names, else it will cause an unknown type error. So this is registering qobjects from extern C++Qt yes